src/index.ts

```diff
@@ -139,8 +144,8 @@ export default class OpentracingExtension<TContext extends SpanContext>
       childOf: externalSpan ? externalSpan : undefined
     });
 
-    rootSpan.setTag("queryString", infos.queryString);
-    this.requestSpan = rootSpan;
+    //rootSpan.setTag("queryString", this.processSpan(infos.queryString? infos.queryString.toString() : ""));
+    this.requestSpan = this.processSpan(rootSpan, infos);
```

Comment:
I was thinking more in this direction, so that the library user does not need to return a specific thing from this hook.

```suggestion
	this.onRequestResolve(rootSpan, infos);
    this.requestSpan = rootSpan;
```


Reading your implementation I wonder what returning a span here helps, could you give me more context here?

Reply:
You're right. It doesn't make sense.
I pushed changes.